File: src/ui/src/main/java/com/amazon/sample/ui/web/payload/CartItem.java
package com.amazon.sample.ui.web.payload;

import com.amazon.sample.ui.clients.carts.model.Item;
import com.amazon.sample.ui.clients.catalog.model.ModelProduct;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class CartItem {
    private String id;

    private int quantity;

    private int price;

    private String name;

    private String imageUrl;

    public static CartItem from(Item item, ModelProduct product) {
        return new CartItem(product.getId(),
                item.getQuantity(), product.getPrice(),
                product.getName(), product.getImageUrl());
    }

    public int getTotalPrice() {
        return this.quantity * this.price;
    }
}

File: src/e2e/pages/CartItem.js
class CartItem {

  constructor(element) {
    this.element = element;
  }

  getName() {
    return this.element.element(by.css('.item-name')).getText();
  }

  getPrice() {
    return this.element.element(by.css('.item-price')).getText();
  }

  getQuantity() {
    return this.element.element(by.css('.item-quantity')).getText();
  }

  async remove() {
    await this.element.element(by.css('.remove-item')).click()
  }
}
module.exports = CartItem;